src/libLocalDescriptorCache/DescDbProxy.cpp

```diff
@@ -352,13 +352,12 @@ void DescDbProxy::getDescriptors(const std::string &f_hostAddress)
 
 grpc::Status DescDbProxy::closeDescDbStream(std::optional<std::chrono::time_point<std::chrono::system_clock>> deadline)
 {
-    if( m_disableCache  == false )//cache enabled, no reflection stream required.
-    {
-        return grpc::Status::OK;
-    }
-
     if ( m_reflectionDescDb == nullptr )
     {
+        if( m_disableCache  == false )//cache enabled, no reflection stream required.
```

Comment:
What about just remove the check and `return` once `m_reflectionDescDb == nullptr ` yields true? (meaning replace whe whole outer `if` body with a `return;`)

Sotty github woudn't let me write a suggestion for this.

Reply:
agrred, the unique ptr == null garauntees the stream is atready closed.